newrelic/hooks/external_botocore.py

```diff
@@ -1249,6 +1273,54 @@ def wrap_serialize_to_request(wrapped, instance, args, kwargs):
     ("kinesis", "get_records"): aws_message_trace(
         "Consume", "Stream", extract_kinesis, extract_agent_attrs=extract_kinesis_agent_attrs, library="Kinesis"
     ),
+    ("firehose", "create_delivery_stream"): aws_function_trace(
+        "create_delivery_stream", extract_firehose, extract_agent_attrs=extract_firehose_agent_attrs, library="Firehose"
+    ),
+    ("firehose", "delete_delivery_stream"): aws_function_trace(
+        "delete_delivery_stream", extract_firehose, extract_agent_attrs=extract_firehose_agent_attrs, library="Firehose"
+    ),
+    ("firehose", "describe_delivery_stream"): aws_function_trace(
+        "describe_delivery_stream",
+        extract_firehose,
+        extract_agent_attrs=extract_firehose_agent_attrs,
+        library="Firehose",
+    ),
+    ("firehose", "list_delivery_streams"): aws_function_trace(
+        "list_delivery_streams", extract_firehose, extract_agent_attrs=extract_firehose_agent_attrs, library="Firehose"
+    ),
+    ("firehose", "list_tags_for_delivery_stream"): aws_function_trace(
+        "list_tags_for_delivery_stream",
+        extract_firehose,
+        extract_agent_attrs=extract_firehose_agent_attrs,
+        library="Firehose",
+    ),
+    ("firehose", "start_delivery_stream_encryption"): aws_function_trace(
+        "start_delivery_stream_encryption",
+        extract_firehose,
+        extract_agent_attrs=extract_firehose_agent_attrs,
+        library="Firehose",
+    ),
+    ("firehose", "stop_delivery_stream_encryption"): aws_function_trace(
+        "stop_delivery_stream_encryption",
+        extract_firehose,
+        extract_agent_attrs=extract_firehose_agent_attrs,
+        library="Firehose",
+    ),
+    ("firehose", "tag_delivery_stream"): aws_function_trace(
+        "tag_delivery_stream", extract_firehose, extract_agent_attrs=extract_firehose_agent_attrs, library="Firehose"
+    ),
+    ("firehose", "untag_delivery_stream"): aws_function_trace(
+        "untag_delivery_stream", extract_firehose, extract_agent_attrs=extract_firehose_agent_attrs, library="Firehose"
+    ),
+    ("firehose", "update_destination"): aws_function_trace(
+        "update_destination", extract_firehose, extract_agent_attrs=extract_firehose_agent_attrs, library="Firehose"
+    ),
+    ("firehose", "put_record"): aws_message_trace(
```

Comment:
According to the [spec](https://source.datanerd.us/agents/agent-specs/blob/main/implementation_guides/aws-sdk.md#kinesis-firehose) we aren't supposed to create message traces for these. I was also inclined to add them as message traces but we aren't supposed to-see comment [here](https://source.datanerd.us/agents/agent-specs/pull/706/files#r826578).

Reply:
That seems bizarre but we can go with that for now